### PR TITLE
[INLONG-2440][CI] Add a workflow to Scan InLong common vulnerabilities

### LINKED ISSUE #2440: [Improve] add a workflow to Scan InLong common vulnerabilities
### Description

add a workflow to Scan InLong common vulnerabilities

### InLong Component

Other for not specified component

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dockerzhang**
### Title Name: [INLONG-2440][CI] Add a workflow to Scan InLong common vulnerabilities

Fixes #2440

### Motivation

Add a workflow to Scan InLong common vulnerabilities.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (shink) [inline .github/workflows/codeql-analysis.yml:42]: Now, the latest checkout action version is `v3`.
```suggestion
      - name: Checkout
        uses: actions/checkout@v3
```

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: ${{ matrix.language }}
```
Comment (shink) [inline .github/workflows/codeql-analysis.yml:48]: Now, the latest codeql-action version is v2.
```suggestion
      - name: Initialize CodeQL
        uses: github/codeql-action/init@v2
        with:
          languages: ${{ matrix.language }}
```

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+      # build project
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: 8
+          distribution: adopt
```
Comment (shink) [inline .github/workflows/codeql-analysis.yml:55]: Now, the latest setup-java action version is v3.
```suggestion
      - name: Set up JDK
        uses: actions/setup-java@v3
        with:
          java-version: 8
          distribution: adopt
```

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository
+            !~/.m2/repository/org/apache/inlong
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```
Comment (shink) [inline .github/workflows/codeql-analysis.yml:64]: Now, the latest cache action version is v3.
```suggestion
      - name: Cache Maven packages
        uses: actions/cache@v3
        with:
          path: |
            ~/.m2/repository
            !~/.m2/repository/org/apache/inlong
          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
          restore-keys: ${{ runner.os }}-m2
```

File: .github/workflows/codeql-analysis.yml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+# limitations under the License.
+#
+
+name: InLong Vulnerabilities Scan
+
+on:
+  push:
+    branches: [ master, 'INLONG-*' ]
+  pull_request:
+    branches: [ master, 'INLONG-*' ]
```
Comment (shink) [inline .github/workflows/codeql-analysis.yml:24]: This check can be triggered on another branch, such as `release`. And we can add configuration in `.github/codeql/codeql-config.yml`. Please check out https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-a-custom-configuration-file

```suggestion
on: [ push, pull_request ]
```

**Comment by shink** [comment]
@dockerzhang We can use an customed configuration file and then disable the default queries. Please check out https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#disabling-the-default-queries
But I don't test it. I can test it latter.

**Comment by shink** [comment]
@dockerzhang Unfortunately, there should be at least one query. Please see this [check](https://github.com/apache/incubator-inlong/runs/6786713924?check_suite_focus=true).

```
Error: Did not detect any queries to run for java. Please make sure that the default queries are enabled, or you are specifying queries to run.
```

File: .github/workflows/ci_codeql.yml [hunk 6]
```diff
@@ -24,8 +24,6 @@ jobs:
     name: Analyze by CodeQL
     runs-on: ubuntu-latest
     permissions:
-      actions: read
-      contents: read
```
Comment (shink) [inline .github/workflows/ci_codeql.yml]: These two fields are only required for workflows in private repositories. So I deleted them.

Comment (dockerzhang) [reply .github/workflows/ci_codeql.yml]: change back to `disable-default-queries: false`  ?

**Comment by shink** [comment]
@dockerzhang Does this [dependency-review-action](https://github.com/actions/dependency-review-action) meet your need? This Action can detect vulnerable dependencies in PRs.

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by shink — APPROVED** [review]

**Review by woofyzhao — APPROVED** [review]

**Pull request merged**